thrust/thrust/iterator/counting_iterator.h

```diff
@@ -156,14 +160,17 @@ struct make_counting_iterator_base
 //!
 //! \see make_counting_iterator
 template <typename Incrementable,
-          typename System     = use_default,
-          typename Traversal  = use_default,
-          typename Difference = use_default>
+          typename System       = use_default,
+          typename Traversal    = use_default,
+          typename Difference   = use_default,
+          typename StrideHolder = detail::empty>
```

Comment:
Nitpick: couldnt this also just be `integral_constant<size_t, 1>`

Reply:
I redefined as:
```c++
using no_stride = compile_time_value<1>;
```
I cannot use `integral_constant` becase:
```c++
// we cannot use ::cuda::std::integral_constant, because it has a conversion operator to T that causes an ambiguity
// with operator+(counting_iterator, counting_iterator::difference_type) in any expression `counting_iterator +
// integral`.
```

Reply:
And I renamed `no_stride` to `unit_stride`.